src/Client/Milestone/MilestoneClient.php

```diff
@@ -0,0 +1,203 @@
+     *
+     * @return ResourceInformationInterface
+     * @throws EmptyResultException
+     */
+    public function getByExternalId(string $id): ResourceInformationInterface
+    {
+        return $this->responseMapper->getObject(
+            $this->soapClient->getByExternalId(['external_id' => $id]),
+            'getByExternalIdResult',
+            $this->hydratorConfigFactory->createResourceInformationConfig()
+        );
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return ResourceInformationInterface
+     * @throws EmptyResultException
+     */
+    public function getById(int $id): ResourceInformationInterface
+    {
+        return $this->responseMapper->getObject(
+            $this->soapClient->getById(['resource_id' => $id]),
+            'getByIdResult',
+            $this->hydratorConfigFactory->createResourceInformationConfig()
+        );
+    }
+
+    /**
+     * @param int $id
+     * @param int $mandatorId
+     *
+     * @return ResourceInformationInterface[]
+     * @throws EmptyResultException
+     */
+    public function getByTypeId(int $id, int $mandatorId): array
+    {
+        return $this->responseMapper->getObjects(
+            $this->soapClient->getByTypeId(['type_id' => $id, 'mandator_id' => $mandatorId]),
+            'getByTypeIdResult',
+            $this->hydratorConfigFactory->createResourceInformationConfig()
+        );
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return FolderInformationInterface
+     * @throws EmptyResultException
+     */
+    public function getDefaultFolderByMandatorId(int $id): FolderInformationInterface
+    {
+        return $this->responseMapper->getObject(
+            $this->soapClient->getResourceDefaultCategory(['customer_id' => $id]),
+            'getResourceDefaultCategoryResult',
+            $this->hydratorConfigFactory->createFolderInformationConfig()
+        );
+    }
+
+    /**
+     *
+     * @return ResourceTypeInformationInterface[]
+     * @throws EmptyResultException
+     */
+    public function getTypeIds(): array
+    {
+        return $this->responseMapper->getObjects(
+            $this->soapClient->getTypeIds(),
+            'getTypeIdsResult',
+            $this->hydratorConfigFactory->createResourceTypeInformationConfig()
+        );
+    }
+
+    /**
+     * @param int $id
+     * @param int $folderId
+     *
+     * @return ResourceInformationInterface
+     * @throws EmptyResultException
+     */
+    public function move(int $id, int $folderId): ResourceInformationInterface
+    {
+        return $this->responseMapper->getObject(
+            $this->soapClient->move(['resource_id' => $id, 'category_id' => $folderId]),
+            'moveResult',
+            $this->hydratorConfigFactory->createResourceInformationConfig()
+        );
+    }
+
+    /**
+     *
+     * @return ServiceStatusInterface
+     * @throws EmptyResultException
+     */
+    public function getServiceAvailable(): ServiceStatusInterface
```

Comment:
If we could implement this part in general - otherwise we would have to do the same for every single client as well

Reply:
Thanks for the annotations! I didn't realize there is a trait for these methods